samples/MvcSandbox/Models/TestModel.cs

```diff
@@ -1,13 +1,18 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.ComponentModel.DataAnnotations;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
 
 namespace MvcSandbox
 {
     public class TestModel : PageModel
     {
+        [FromQuery]
+        [Required]
```

Comment:
We should not support `[Required]` on a property. It doesn't work with controllers or parameters. We have no way for you to add a validator to a top level object via an attribute